src/dra/4A538.c

```diff
@@ -805,125 +802,125 @@ void RenderEntities(void) {
         if (r->y < -512 || r->y > 512) {
             continue;
         }
-        temp_a0_2 = r->animCurFrame & 0x7FFF;
-        if (temp_a0_2 == 0) {
+        animFrameIdx = r->animCurFrame & 0x7FFF;
+        if (!animFrameIdx) {
             continue;
         }
         if (r->animSet > 0) {
-            var_t2 = D_800A3B70[r->animSet][temp_a0_2];
+            animFrame = D_800A3B70[r->animSet][animFrameIdx];
         } else {
-            spriteBank = g_api.o.spriteBanks;
+            spriteBank = (s16**)&g_api.o.spriteBanks[0];
             spriteBank = &spriteBank[r->animSet & 0x7FFF];
-            spriteBank = (s16**)*spriteBank;
-            var_t2 = spriteBank[temp_a0_2];
+            spriteBank = (s16**)spriteBank[0];
+            animFrame = spriteBank[animFrameIdx];
         }
-        r->unk1C = *var_t2++;
-        if (r->unk1C & 0x8000) {
+        r->spriteSheetIdx = *animFrame++;
+        if (r->spriteSheetIdx & 0x8000) {
             plDraw = &g_PlayerDraw[entity->unk5A];
-            r->unk20 = plDraw->flipX ^ r->unk20;
-            r->unk1C &= 0x7FFF;
+            r->flipX ^= (u16)plDraw->flipX;
+            r->spriteSheetIdx &= 0x7FFF;
             if (r->animSet > 0) {
-                if (r->animSet == 13) { // BAT FORM
-                    stuff = r->unk1C + g_PlOvlAluBatSpritesheet[0];
-                    spriteData = *stuff;
+                if (r->animSet == 13) {
+                    spriteData = (*g_PlOvlAluBatSpritesheet)[r->spriteSheetIdx];
                 } else {
-                    stuff = r->unk1C + g_PlOvlSpritesheet;
-                    spriteData = *stuff;
+                    spriteData = ((u8**)SPRITESHEET_PTR)[r->spriteSheetIdx];
```

Comment:
I know it is not part of your PR, but `SPRITESHEET_PTR` in `game.h` at line `214` needs to be changed to `#define SPRITESHEET_PTR g_PlOvlSpritesheet`, otherwise the game will crash. I am still investigating why this PR does not render the Alucard sprite.

Reply:
Done!